### ISSUE TITLE
IOError: cannot identify image file

### ISSUE THREAD

**Issue by vinczente** [thumbor/thumbor#538]
Hi there,

While generating some thumbnails I came across this error.
```
2015-08-19 06:58:55 thumbor:ERROR ERROR: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/tornado-3.2.2-py2.7-linux-x86_64.egg/tornado/web.py", line 1320, in _stack_context_handle_exception
    raise_exc_info((type, value, traceback))
  File "/usr/local/lib/python2.7/dist-packages/tornado-3.2.2-py2.7-linux-x86_64.egg/tornado/stack_context.py", line 302, in wrapped
    ret = fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/thumbor-4.7.1-py2.7-linux-x86_64.egg/thumbor/loaders/http_loader.py", line 49, in return_contents
    callback(response.body)
  File "/usr/local/lib/python2.7/dist-packages/thumbor-4.7.1-py2.7-linux-x86_64.egg/thumbor/handlers/__init__.py", line 319, in handle_loader_loaded
    self.context.request.engine.load(buffer, extension)
  File "/usr/local/lib/python2.7/dist-packages/thumbor-4.7.1-py2.7-linux-x86_64.egg/thumbor/engines/__init__.py", line 85, in load
    imageOrFrames = self.create_image(buffer)
  File "/usr/local/lib/python2.7/dist-packages/thumbor-4.7.1-py2.7-linux-x86_64.egg/thumbor/engines/pil.py", line 50, in create_image
    img = Image.open(BytesIO(buffer))
  File "build/bdist.linux-x86_64/egg/PIL/Image.py", line 2028, in open
    raise IOError("cannot identify image file")
IOError: cannot identify image file
```

```
Pillow==2.3.2
libthumbor==1.1.0
thumbor==4.7.1
thumbor-pexif==0.14.1
```

curl "http://127.0.0.1:8888/unsafe/750x421/center/top/smart/http%3A%2F%2Fcps-static.rovicorp.com%2F2%2FOpen%2FNickelodeon%2FPeter%2520Rabbit%2Fshowchoriz.jpg"

curl "http://127.0.0.1:8888/unsafe/375x750/center/top/smart/http%3A//cps-static.rovicorp.com/2/Open/MTV/Randy%2520Jackson%2520Presents%2520-%2520America's%2520Best%2520Dance%2520Crew/Judges%2520JC-Lil%2520Mama-Shane%2520of%2520ABDC%2520(2).jpg"

Can you replicate this ? my Thumbor throws a 500 error

**Comment by masom** [comment]
This also happens on 
thumbor==5.0.6
tornado==4.2.1
Pillow==2.9.0

**Comment by masom** [comment]
Running
`identify showchoriz.jpg`

Produces the following output:
```
showchoriz.jpg JPEG 4800x3600 4800x3600+0+0 8-bit sRGB 2.275MB 0.000u 0:00.000
identify: Corrupt JPEG data: 9672 extraneous bytes before marker 0xee `showchoriz.jpg' @ warning/jpeg.c/JPEGWarningHandler/352.
```

**Comment by masom** [comment]
There's a StackOverflow answer for something similar : http://stackoverflow.com/questions/22610684/reading-a-jpeg-in-python-pil-with-broken-header

If you have control over that JPEG you should fix the image or fill a bug report with Pillow.

**Comment by vinczente** [comment]
Thanks @masom 
I dont really care about the picture resizing failing but thumbor seems to hang after, this might be the reason:
https://github.com/thumbor/thumbor/issues/528

**Comment by masom** [comment]
@vinczente Yep, the hanging is definitely a bug. If an exception is raised within one of the tornado generator the exception does not bubble up properly and nothing happens.

**Comment by masom** [comment]
https://github.com/thumbor/thumbor/pull/541 fixes this.

**Comment by heynemann** [comment]
Already merged #541.

Thanks guys for reporting.
